src/compiler/scala/tools/nsc/transform/patmat/Logic.scala

```diff
@@ -230,7 +230,8 @@ trait Logic extends Debugging {
           checkPair(a, b) || checkPair(a, c) || checkPair(b, c)
         } else {
           val ops = new Array[Prop](size)
-          ops0.copyToArray(ops)
+          val copied = ops0.copyToArray(ops)
+          assert(copied == ops.length, "")
```

Comment:
I wonder about the new assertions after `copyToArray`. They are of the shape

```
val a = new Array[T](xs.length)
val cp = xs.copyToArray(a)
assert(cp == xs.length)
```

the assertion seems mostly redundant.

Reply:
I was also a bit dubious, but sometimes it relies on an iterator being consistent with `knownSize`, or some such invariant. Although the assertion might be trivial (or worse, not no-cost), my other concern was that it might be non-trivial and actually throw in some edge case I was unable to see. (I did not audit this idiom.)

Reply:
Sample non-trivial copy: `e.newText.copyToArray(patchedChars, outNew)`

Reply:
Some of the new assertions are not trivial, and they are actually the ones that make me uncomfortable... We could introduce a buggy assertion.

Reply:
I will comment them out and return to the question later. The dilemma is that if an assertion correctly fails, is it desirable? "My buggy code worked just fine."